### ISSUE TITLE
Add stable expression representation

### ISSUE THREAD

**Issue by xFrednet** [rust-marker/marker#52]
Create an initial version of the stable expression representation.

* Ungrouped expressions
    * [x] [LiteralExpression](https://doc.rust-lang.org/reference/expressions/literal-expr.html)
    * [x] [PathExpression](https://doc.rust-lang.org/reference/expressions/path-expr.html)
    	* https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=295638462fb52346cd02afbe7bf3693c
    * [x] [OperatorExpression](https://doc.rust-lang.org/reference/expressions/operator-expr.html)
        * [x] AssignOperationExpressions
    * [x] [BlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html)
        * [x] [AsyncBlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html#async-blocks)
        * [x] [UnsafeBlockExpression](https://doc.rust-lang.org/reference/expressions/block-expr.html#unsafe-blocks)
    * [x] [ClosureExpression](https://doc.rust-lang.org/reference/expressions/closure-expr.html)
    * [ ] [AwaitExpression](https://doc.rust-lang.org/reference/expressions/await-expr.html) https://github.com/rust-marker/marker/issues/174
* Conditional expressions
    * [x] [IfExpression](https://doc.rust-lang.org/reference/expressions/if-expr.html#if-expressions)
    * [x] [IfLetExpression](https://doc.rust-lang.org/reference/expressions/if-expr.html#if-let-expressions)
    * [x] [MatchExpression](https://doc.rust-lang.org/reference/expressions/match-expr.html)
* Call expressions
    * [x] [CallExpression](https://doc.rust-lang.org/reference/expressions/call-expr.html)
    * [x] [MethodCallExpression](https://doc.rust-lang.org/reference/expressions/method-call-expr.html)
* Loop expressions
    * [x] [LoopExpression](https://doc.rust-lang.org/reference/expressions/loop-expr.html)
    * [x] [ContinueExpression](https://doc.rust-lang.org/reference/expressions/loop-expr.html#continue-expressions)
    * [x] [BreakExpression](https://doc.rust-lang.org/reference/expressions/loop-expr.html#break-expressions)
    * [x] [ReturnExpression](https://doc.rust-lang.org/reference/expressions/return-expr.html)
* Constructor expressions
    * [x] [ArrayExpression](https://doc.rust-lang.org/reference/expressions/array-expr.html)
    * [x] [RangeExpression](https://doc.rust-lang.org/reference/expressions/range-expr.html)
    * [x] [TupleExpression](https://doc.rust-lang.org/reference/expressions/tuple-expr.html)
    * [x] [StructExpression](https://doc.rust-lang.org/reference/expressions/struct-expr.html)
* Place expressions
    * [x] [IndexExpression](https://doc.rust-lang.org/reference/expressions/array-expr.html#array-and-slice-indexing-expressions)
    * [x] [TupleIndexingExpression](https://doc.rust-lang.org/reference/expressions/tuple-expr.html#tuple-indexing-expressions)
    * [x] [FieldExpression](https://doc.rust-lang.org/reference/expressions/field-expr.html)
* Syntax expressions (Probably won't be added)
    * [ ] [GroupedExpression](https://doc.rust-lang.org/reference/expressions/grouped-expr.html)
    * [ ] [UnderscoreExpression](https://doc.rust-lang.org/reference/expressions/underscore-expr.html)

See: <https://doc.rust-lang.org/reference/expressions.html>

---

This will be another big one. On the other hand, this'll probably be the final stretch before v0.0.1

**Comment by xFrednet** [comment]
This is kind of blocked on #50

**Comment by xFrednet** [comment]
The two ticked off items, will be added in #82 along a basic statement representation. The literal expression has been split up, to handle the specific types explicitly :)

**Comment by xFrednet** [comment]
I just remembered that expressions will also require the translation of semantic types to Marker types. Add that to the pile of open todos :sweat_smile:

**Comment by xFrednet** [comment]
Most expressions are done now. I'm closing this in favor of https://github.com/rust-marker/marker/issues/174 which tracks the last expression which will need to be added to the API :)
